### PR TITLE
Fix lib/autoload.php

### PR DESCRIPTION
**Pull request by Art4**
This PR fixes an autoloading bug if composer isn't used. It was introduced with #203.

I've also moved the tests in a more logically folder structure and moved them into proper namespace.

- unit tests are now in `\Redmine\Tests\Unit`
- fixtures are now in `\Redmine\Tests\Fixtures`

This allows us to simplify the autoloading because of PSR-4. And we have now the option to create other tests like integration tests in the `\Redmine\Tests\Integration` namespace.

Btw: [I was wondering](https://github.com/kbsali/php-redmine-api/pull/203#issue-231045169) how Travix could load the PHPUnit classes. Turns out PHPUnit binary looks automatically for the composer `vender/autoload.php` and includes it, but I can't find this behavior in the PHPUnit docs.

### CODE REVIEW

File: lib/autoload.php [hunk 1]
```diff
@@ -1,15 +1,51 @@
 <?php
 
+// use Composer if possible
```
Comment (kbsali) [inline lib/autoload.php]: @Art4 anyway you could cs fix this? (and var should be camel cased! ;) )

Comment (Art4) [inline lib/autoload.php]: Done

File: lib/autoload.php [hunk 2]
```diff
@@ -1,15 +1,46 @@
...
-     * to map directly to a location in the filesystem.
-     */
-    $class = str_replace('\\', '/', $class);
-
-    // Check under the current directory
-    $path = dirname(__FILE__).'/'.$class.'.php';
-    if (file_exists($path)) {
-        require_once $path;
+    // project-specific namespace prefix and base directory with trailing /
+    $namespace_map = [
```
Comment (kbsali) [inline lib/autoload.php]: camelCase please :)

File: lib/autoload.php [hunk 3]
```diff
@@ -1,15 +1,46 @@
...
+        'Redmine\\'        => __DIR__.'/Redmine/',
+        'Redmine\\Tests\\' => __DIR__.'/../tests/',
+    ];
+
+    foreach ($namespace_map as $prefix => $base_dir) {
```
Comment (kbsali) [inline lib/autoload.php]: here too

File: lib/autoload.php [hunk 4]
```diff
@@ -1,15 +1,46 @@
...
+        // does the class use the namespace prefix?
+        $len = strlen($prefix);
+        if (strncmp($prefix, $class, $len) !== 0) {
+            // no, move to the next registered autoloader
+            continue;
+        }
+
+        // get the relative class name
+        $relative_class = substr($class, $len);
```
Comment (kbsali) [inline lib/autoload.php]: and there

File: lib/autoload.php [hunk 5]
```diff
@@ -1,15 +1,46 @@
...
+
+        // replace the namespace prefix with the base directory, replace namespace
+        // separators with directory separators in the relative class name, append
+        // with .php
+        $file = $base_dir.str_replace('\\', '/', $relative_class).'.php';
```
Comment (Art4) [inline lib/autoload.php]: And there 😉

**Comment by Art4** [comment]
All var names should be fixed now. 😁

**Comment by kbsali** [comment]
:clap: thanks!

**Pull request merged**